apis/src/components/organisms/quickplay.rs

```diff
@@ -0,0 +1,83 @@
+use core::panic;
+use leptos::*;
+
+use leptos_router::use_navigate;
+
+use crate::{
+    common::ChallengeAction,
+    providers::{ApiRequests, AuthContext},
+};
+use hive_lib::{ColorChoice, GameType};
+use shared_types::ChallengeVisibility;
+use shared_types::{ChallengeDetails, TimeMode};
+
+#[component]
+fn GridButton(text: String) -> impl IntoView {
+    let display_text = text.clone();
+    view! {
+        <button
+            class="px-4 py-2 font-bold text-white rounded bg-button-dawn dark:bg-button-twilight hover:bg-pillbug-teal active:scale-95"
+            on:click=move |_| {
+                let api = ApiRequests::new();
+                let auth_context = expect_context::<AuthContext>();
+                let account = match (auth_context.user)() {
+                    Some(Ok(Some(account))) => Some(account),
+                    _ => None,
+                };
+                let (base, increment) = match text.as_str() {
+                    "1+0" => (1, 0),
+                    "2+1" => (2, 1),
+                    "3+0" => (3, 0),
+                    "3+2" => (3, 2),
+                    "5+0" => (5, 0),
+                    "5+3" => (5, 3),
+                    "10+0" => (10, 0),
+                    "10+5" => (10, 5),
+                    "15+10" => (15, 10),
+                    _ => panic!("Invalid text"),
+                };
+                if account.is_some() {
+                    let details = ChallengeDetails {
+                        rated: false,
```

Comment:
Rated needs to be true here.

Reply:
made rated true but also set the ELO upper bound to +300 since auto matching will happen. let me know if this should be adjusted

Reply:
Leave it None for now, otherwise lgtm. Let's see what leex says